lib/puppet/pops/serialization/to_data_converter.rb

```diff
@@ -108,6 +108,10 @@ def to_data(value)
         process(value) do
           { PCORE_TYPE_KEY => PCORE_TYPE_SENSITIVE, PCORE_VALUE_KEY => to_data(value.unwrap) }
         end
+      elsif value.instance_of?(Puppet::Pops::Types::TypeFactory.deferred.implementation_class)
+        process(value) do
+          { PCORE_TYPE_KEY => PCORE_TYPE_DEFERRED, 'name' => value.name, 'arguments' => value.arguments }
+        end
```

Comment:
I don't think this is needed now since Deferred is an Object. When I tried just calling the to_data_converter I get the expected hash with `{__ptype: "Deferred", ... }`

Reply:
ha, but not when @rich_data is false

Reply:
Without this code, I get the warning that it is stringifying the deferred class when the `unknown_to_data(value)` method runs during an agent run, specifically when it is caching the catalog.

Reply:
Are you sure `rich_data` is on when caching the catalog? If not it will destroy a catalog with rich data.
The method does this:
```
def unknown_to_data(value)
  @rich_data ? value_to_data_hash(value) : unknown_to_string_with_warning(value)
end
```
So, it should just convert the object to its data hash.

I am not sure why the special handling of Sensitive is there - it seems to always produce a hash with __ptype => Sensitive even if not using rich_data.